Fix Packaging.Jenkinsfile

ionutbalutoiu opened the pull request:
Use the `vcvars64.bat` script instead of the `LaunchDevCmd.bat` to correctly
prepare the cmd shell for Visual Studio usage.

Also, `cmake` is available only after the VS environment is set (either via `vcvars64.bat` or `LaunchDevCmd.bat`)

ionutbalutoiu commented:
This fixes some recent failures with the OE nightly packages Jenkins job.

ionutbalutoiu commented:
bors r+

aserdean commented:
/LGTM

Maybe also add :
`
--output-on-failure
Output anything outputted by the test program if the test should fail. This option can also be enabled by setting the environment variable CTEST_OUTPUT_ON_FAILURE`
to ctest?

ionutbalutoiu commented:
> /LGTM
> 
> Maybe also add :
> ` --output-on-failure Output anything outputted by the test program if the test should fail. This option can also be enabled by setting the environment variable CTEST_OUTPUT_ON_FAILURE`
> to ctest?

Thanks for the suggestion @aserdean. I just removed the `ctest` invocation since this stage is used only to collect the `build/tools/oeedger8r/oeedger8r.exe` binary after a successful build.

I also switched to the Ninja build since it uses the full CPU for building.

ionutbalutoiu commented:
Bors doesn't execute this `Packaging.Jenkinsfile`.

However, I manually run this against my PR code. This is a successful validation from the CI:

https://oe-jenkins.eastus.cloudapp.azure.com/blue/organizations/jenkins/OpenEnclave-nightly_packages-ibalutoiu-temp/detail/OpenEnclave-nightly_packages-ibalutoiu-temp/5/pipeline/

aserdean commented:
> > /LGTM
> > Maybe also add :
> > ` --output-on-failure Output anything outputted by the test program if the test should fail. This option can also be enabled by setting the environment variable CTEST_OUTPUT_ON_FAILURE`
> > to ctest?
> 
> Thanks for the suggestion @aserdean. I just removed the `ctest` invocation since this stage is used only to collect the `build/tools/oeedger8r/oeedger8r.exe` binary after a successful build.
> 
> I also switched to the Ninja build since it uses the full CPU for building.

Alternately you can do the following:
```
cmake -G "Visual Studio 15 2017 Win64"
msbuild tools\oeedger8r\oeedger8r_target.vcxproj -p:Configuration=Debug
```

ionutbalutoiu commented:
> > > /LGTM
> > > Maybe also add :
> > > ` --output-on-failure Output anything outputted by the test program if the test should fail. This option can also be enabled by setting the environment variable CTEST_OUTPUT_ON_FAILURE`
> > > to ctest?
> > 
> > 
> > Thanks for the suggestion @aserdean. I just removed the `ctest` invocation since this stage is used only to collect the `build/tools/oeedger8r/oeedger8r.exe` binary after a successful build.
> > I also switched to the Ninja build since it uses the full CPU for building.
> 
> Alternately you can do the following:
> 
> ```
> cmake -G "Visual Studio 15 2017 Win64"
> msbuild tools\oeedger8r\oeedger8r_target.vcxproj -p:Configuration=Debug
> ```

Thanks @aserdean! Works as a charm.

ionutbalutoiu commented:
And [here](https://oe-jenkins.eastus.cloudapp.azure.com/blue/organizations/jenkins/OpenEnclave-nightly_packages/detail/OpenEnclave-nightly_packages/198/pipeline) is the CI validation that the changes from this PR are fine.

We're good to merge now.

bors r+
